ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/ProcessingLogServerUtils.java

```diff
@@ -0,0 +1,101 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.util;
+
+import io.confluent.common.logging.LogRecordStructBuilder;
+import io.confluent.ksql.exception.KafkaTopicExistsException;
+import io.confluent.ksql.function.InternalFunctionRegistry;
+import io.confluent.ksql.metastore.MetaStoreImpl;
+import io.confluent.ksql.parser.KsqlParser;
+import io.confluent.ksql.parser.KsqlParser.PreparedStatement;
+import io.confluent.ksql.parser.SqlFormatter;
+import io.confluent.ksql.parser.tree.AbstractStreamCreateStatement;
+import io.confluent.ksql.processing.log.ProcessingLogMessageFactory;
+import io.confluent.ksql.services.KafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.TypeUtil;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class ProcessingLogServerUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ProcessingLogServerUtils.class);
+
+  private ProcessingLogServerUtils() {
+  }
+
+  public static Schema getMessageSchema() {
+    return new LogRecordStructBuilder()
+        .withMessageSchemaAndValue(
+            new SchemaAndValue(ProcessingLogMessageFactory.PROCESSING_LOG_SCHEMA, null))
+        .build()
+        .schema();
+  }
+
+  private static String getTopicName(
+      final ProcessingLogConfig config,
+      final KsqlConfig ksqlConfig) {
+    final String topicNameConfig = config.getString(ProcessingLogConfig.TOPIC_NAME);
+    if (topicNameConfig.equals(ProcessingLogConfig.TOPIC_NAME_NOT_SET)) {
+      return String.format(
+          "%s%s",
+          ksqlConfig.getString(KsqlConfig.KSQL_SERVICE_ID_CONFIG),
+          ProcessingLogConfig.TOPIC_NAME_DEFAULT_SUFFIX
+      );
+    } else {
+      return topicNameConfig;
+    }
+  }
+
+  public static void maybeCreateProcessingLogTopic(
+      final KafkaTopicClient topicClient,
+      final ProcessingLogConfig config,
+      final KsqlConfig ksqlConfig) {
+    if (!config.getString(ProcessingLogConfig.TOPIC_AUTO_CREATE).equals(
+        ProcessingLogConfig.AUTO_CREATE_ON)) {
+      return;
+    }
+    final String topicName = getTopicName(config, ksqlConfig);
+    final int nPartitions =
+        config.getInt(ProcessingLogConfig.TOPIC_PARTITIONS);
+    final short nReplicas =
+        config.getShort(ProcessingLogConfig.TOPIC_REPLICATION_FACTOR);
+    try {
+      topicClient.createTopic(topicName, nPartitions, nReplicas);
+    } catch (final KafkaTopicExistsException e) {
+      LOGGER.info(String.format("Log topic %s already exists", topicName), e);
+    }
+  }
+
+  public static String processingLogStreamCreateStatement(
```

Comment:
This looks to build a string, to parse it, to rebuild it, to return it. The only prod code calling this then parses it and executes it.

Why not just build the `CreateStream` object yourself and return it? See `KsqlRestApplication`@332 for an example.

Reply:
Just trying to stick with the property public interface of the parser. It feels like that's not the way those classes are meant to be used, since they are creating these deeply nested objects with all their internal details exposed via the constructors - i.e. the constructors should be package-private and the only thing that _should_ be constructing those is AstBuilder. Seems like something that could get annoying to maintain over time. If I add or change a property of Expression I have to change tons of places throughout the code base. Whereas statement strings have to always work.

Reply:
I hear you. I can understand that view.  But then why not just execute the statement against the engine direct, why the string -> parsed to PreparedStatement -> string -> parsed to PreparedStatement -> execute????

Reply:
Now I see what you meant. Yeah that's a fair point. We can just return a PreparedStatement